Pci - 로또게임

pci2676 opened the pull request:
1. 로또 상금찾기를 enum을 이용해서 구현해보았음

2. View에서 input을 mock하는 방법이 올바른지 잘 모르겠음..

3. LottoList를 내가 원하는 숫자로 만들기 위해 상속을 이용해 FixedLottoList를만들었는데 Setter가 열려서 정말 좋지않은코드인거같음... 테스트를위해 만들었는데..

4. LottoResultDto에 멤버변수로 있는 HashMap이 있는데 이 변수가
public하게 getter로 불러 올수 있음 -> 이런걸 한번더 랩핑해서 맘대로 put,
remove를 할수없게 해주면 그것이 일급 컬렉션이 되는 것인가 싶음.

5. HashMap을 만들면서 느낀건데 makeMap 메소듣를 이용해서 HashMap을만들었는데 private한 세터와 생성자에서 멤버변수에 값을 넣어주는 것이 큰 차이가 있는것인지 생각이 듬 --> 이러한 생각때문에 FixedLottoList를 위한 protect한 setter를 일단 만들어서 사용하였음..

Commit:
로또 생성기 구조 구현

pci2676 committed:
난수,고정 숫자 로또 생성

Commit:
로또 구매 구현 완료

pci2676 committed:
로또 당첨 리펙토링중

Commit:
로또 구현중

pci2676 committed:
로또게임
1. 로또 상금찾기를 enum을 이용해서 구현해보았음
2. View에서 input을 mock하는 방법이 올바른지 잘 모르겠음..
3. LottoList를 내가 원하는 숫자로 만들기 위해 상속을 이용해 FixedLottoList를만들었는데 Setter가 열려서 정말 좋지않은코드인거같음... 테스트를위해 만들었는데..
4. LottoResultDto에 멤버변수로 있는 HashMap이 있는데 이 변수가
public하게 getter로 불러 올수 있음 -> 이런걸 한번더 랩핑해서 맘대로 put,
remove를 할수없게 해주면 그것이 일급 컬렉션이 되는 것인가 싶음.
5. HashMap을 만들면서 느낀건데 makeMap 메소듣를 이용해서 HashMap을만들었는데 private한 세터와 생성자에서 멤버변수에 값을 넣어주는 것이 큰 차이가 있는것인지 생각이 듬 --> 이러한 생각때문에 FixedLottoList를 위한 protect한 setter를 일단 만들어서 사용하였음..

pci2676 committed:
LottoList 추상클래스로 변경
기존에 있던 setter를 없애기 위해서 추상클래스 변경하였음
생성자에서 파라미터를 제네릭으로 받아서 처리하였음

pci2676 commented:
기존에 LottoList에 존재하던 setter를 없애기 위해서 LottoList를 추상클래스로 변경해서 실코드에서 사용되는 RandomLottoList와 테스트를 위한 FixedLottoList로 나누어 구현하였음

pci2676 committed:
상수를 보다 직관적인 이름으로 변경하였음

Gobukgol commented on src/main/java/domain/Lotto.java:
```diff
@@ -0,0 +1,39 @@
+                .map(Objects::toString)
+                .collect(Collectors.joining(", ")) + "]";
+    }
+
```
numbers.toString() 하면 [ , , , , , , ,] 요런식으로 출력된다구해!

Gobukgol commented on src/main/java/domain/LottoList.java:
```diff
@@ -0,0 +1,46 @@
+                .collect(Collectors.joining("\n"));
+        System.out.println(list);
+    }
+}
```
.collect 대신에 .forEach로 개행문자 조인시키지않구 출력가능할꺼야.

pci2676 committed:
다시보니까 오버라이드 하는거 하나도없어서 그냥 클래스로 LottoList 바꿈.

pci2676 committed:
컨트롤러 테스트만하고 안만들어놨었음.....

pci2676 committed:
인텔리제이

pci2676 committed:
추가 요구사항 구현

1. LottoNo 클래스로 로또번호들을 감싸 놓았음.
2. WinningLotto클래스로 우승번호와 보너스 번호를 들고있게 하였음
3. 모호하고 직관적이지 않던 메소드명 변수명 변경하였음
4. 결과 출력을 위한 문자열 생성을 OutputView단에서 전부 일임하였음.
5. ManualLottoBundle 클래스에서 수동 및 자동 로또티켓들을 생성함 따라서
기존의 RandomLottoBundle은 테스트 이외에 사용되지 않음

pci2676 committed:
테스트 구분

pci2676 committed:
기존 상금계산 방식을 변경하였음

map을 이용하여 상금을 계산했는데 너무 길고 복잡해서
List에 당첨현황을 담아두고 하나씩 꺼내서 당첨금을 계산하도록 변경하였음

pci2676 committed:
로또 리펙토링

1. 넘버제네레이터를 도메인으로 옮기면서 로또 머신으로 바꿈
2. 로또머신에서 로또 티켓을 발급해줌
3. 로또 번들에선 로또 티켓을 뭉쳐서 보관하도록 하였음
4. 기존 넘버제네레이터는 삭제

pci2676 committed:
LottoBundle클래스를 일급컬렉션으로 만드려고 리펙토링

1. 멤버젼수는 Map으로 로또 수동,랜덤 로또들을 들고 있도록하였음
2. 번들에서 하는 일인데 resultdto에서 들고 있던 메소드를 번들로 옮김

Gobukgol commented on src/main/java/controller/LottoController.java:
```diff
@@ -0,0 +1,47 @@
+        List<ManualNumberDto> manualNumbers = inputView.getManualLottoNumbers(manualAmount);
+
+        InputDto inputDto = new InputDto(price, manualAmount, manualNumbers);
+
```
inputView에서 InputDto 자체를 받는건 어때???

pci2676 committed:
로또 리뷰후 리펙토링

1. exception 테스트 범위제한
2. 공통된 범위의 로또 생성 before로 묶음
3. 수동로또 한장사기 테스트에서 번호도 일치하는지 확인
4. 매번 번호 생성안하고 상수에서 받아다 쓰게 변경
5. LottoResultDto에 불필요한 멤버변수, getter제거
6. 2등로또인지 확인하는 로직을 좀더 명확하게 작성
7. 불필요하게 boolean값을 리턴하던것을 void로 바꿈
8. InputDto를 한번에 받도록 수정하였음
9. 출력을 컨트롤러에서 하던것을 outputview로 이동
10. 컨트롤러단 줄수를 10줄이하로 줄였음

pci2676 committed:
예외처리에 대한 테스트 추가

pci2676 committed:
필요없는 dto지우고 간결하게

1. LottoMachine에서 바로 LottoBundle을 생성하게 하였음

pci2676 committed:
리펙토링

1. 팩토리 메소드 페턴으로 리펙토링
2. dto에게 역할 부여 및 불필요한 dto 제거
Todo
로또 번호들을 vo로 구성하여야함.

pci2676 committed:
로또 리펙토링

1. 로또티켓이 자동인지 수동인지에 대한 여부를 가지고있도록 변경하였음
2. 로또번호를 싱글톤으로 불러오도록 하였음

pci2676 committed:
파라미터를 좀더 읽기 편하게 변경

1. 객체에서 정보를 한번 불러와서 파라미터를 넘겼는데
객체를 넘겨서 해당 메소드에서 풀어서 쓰도록 변경하였음
